### ISSUE TITLE
Breaking change introduced between 3.31.0 and 3.32.0?

### ISSUE THREAD

**Issue by JPLachance** [okta/terraform-provider-okta#1230]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Any supported Terraform versions.

### Affected Resource(s)

The provider configuration now requires new parameters when using an Application client credentials. Before, we had:

```hcl
terraform {
  required_providers {
    okta = {
      source  = "okta/okta"
      version = "3.31.0"
    }
  }
}

[...]

provider "okta" {
  org_name = jsondecode(data.aws_secretsmanager_secret_version.okta_creds.secret_string)["org_name"]
  base_url = jsondecode(data.aws_secretsmanager_secret_version.okta_creds.secret_string)["base_url"]

  client_id   = jsondecode(data.aws_secretsmanager_secret_version.okta_creds.secret_string)["client_id"]
  private_key = jsondecode(data.aws_secretsmanager_secret_version.okta_creds.secret_string)["private_key"]
  scopes      = jsondecode(data.aws_secretsmanager_secret_version.okta_creds.secret_string)["scopes"]
}
```

But with the version 3.32.0 of the provider, this configuration prints us:

> error invalid configuration: when using AuthorizationMode 'PrivateKey', you must supply 'ClientId, 'Scopes', and 'PrivateKey' or 'PrivateKeySigner'.

When we downgrade to 3.31.0, our configuration works, showing the issue is with the new version of the provider.

### Expected Behavior

We did not expect a breaking change in a minor version of this provider. A minor version is supposed to be compatible with the previous one.

### Actual Behavior

Well, we would need to change our provider configuration, everywhere we use the provider.

### Steps to Reproduce

1. `terraform plan`

### References

https://github.com/okta/terraform-provider-okta/compare/v3.31.0...v3.32.0

**Comment by monde** [comment]
@JPLachance this is a regression I will fix and do a new release ASAP{

**Comment by monde** [comment]
#1233 /cc @ericnorris

**Comment by virgofx** [comment]
Just ran into this issue myself and can confirm we have the `scopes`, `private_key`, and `client_id` specified.

**Comment by monde** [comment]
@JPLachance @virgofx fix is out in v3.33.0 now.

**Comment by virgofx** [comment]
Thanks @monde just confirmed it's working on 3.33.0
